### ISSUE TITLE
Modal documentation needs to be updated

### ISSUE THREAD

**Issue by benda-1** [IBM/carbon-components-angular#333]
## Detailed description

 carbon-components-angular/src/modal/modal.component.ts 
line 56 export class SampleModal {

should extends BaseModal and 
line 57 closeModal should be removed as it is inherited from BaseModal
 

this is the working sample code:
```typescript
export class SampleModal extends BaseModal {
  modalText: string;
  constructor(protected injector: Injector) {
    super();
    this.modalText = this.injector.get("modalText");
  }
```

**Comment by youda97** [comment]
Closed by #360

**Comment by youda97** [comment]
Closed by #360
